sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusClientBuilderTest.java

```diff
@@ -44,6 +46,33 @@
         ENDPOINT, SHARED_ACCESS_KEY_NAME, SHARED_ACCESS_KEY, QUEUE_NAME);
     private static final Proxy PROXY_ADDRESS = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(PROXY_HOST, Integer.parseInt(PROXY_PORT)));
 
+    @Test
+    void viaQueueNameWithTopicNotAllowed() {
+        // Arrange & Act
+        ServiceBusSenderClientBuilder builder = new ServiceBusClientBuilder()
+            .connectionString(NAMESPACE_CONNECTION_STRING)
+            .sender()
+            .topicName(TOPIC_NAME)
+            .viaQueueName(VIA_QUEUE_NAME);
+
+        // Assert
+        assertThrows(IllegalStateException.class, () -> builder.buildAsyncClient());
+    }
+
+    @Test
+    void queueClientWithViaQueueName() {
+        // Arrange & Act
+        final ServiceBusSenderAsyncClient client = new ServiceBusClientBuilder()
```

Comment:
This can be split. // Arrange is where the builder is created.

`// Act` is when you do `var client = builder.buildAsyncClient()`